lib/IDE/CodeCompletion.cpp

```diff
@@ -1351,6 +1351,13 @@ class CodeCompletionCallbacksImpl : public CodeCompletionCallbacks {
   void setAttrTargetDeclKind(Optional<DeclKind> DK) override {
     if (DK == DeclKind::PatternBinding)
       DK = DeclKind::Var;
+
+    // If the target is already set to 'Module', that means the completion
+    // should be performed as if it's not tied to any specific decl.
+    // see 'completeDeclAttrBeginning()'
+    if (AttTargetDK == DeclKind::Module)
```

Comment:
I'd prefer we just store the "isindependent" bit, maybe as "AttTargetIsIndependent" so we can just use `None` instead of this dance with `Module`.